cobc/codegen.c

```diff
@@ -1830,14 +1830,28 @@ output_gnucobol_defines (const char *formatted_date)
 	output_line ("#define  COB_MODULE_TIME\t\t%d", i);
 
 	{
-		struct cb_text_list *l = cb_include_file_list ;
-		for (;l;l=l->next){
-			if (l->text[0] == '<'){
+		struct cb_text_list *l = cb_include_file_list;
+		struct cb_text_list *ld = cb_include_file_list_directive;
+		struct cb_text_list *next;
+		for (;l;l=l->next) {
+			if (l->text[0] == '<') {
 				output_line ("#include %s", l->text);
 			} else {
 				output_line ("#include \"%s\"", l->text);
 			}
 		}
+		while (ld) {
+			if (ld->text[0] == '<') {
+				output_line ("#include %s", ld->text);
+			} else {
+				output_line ("#include \"%s\"", ld->text);
+			}
+			next = ld->next;
+			if (next != NULL)
+				free (ld);
```

Comment:
when using `cobc_strdup` and friends, use `cobc_free` here - but shouldn't that be done on both the text (after output) and the list item? Why not doing that for the last?

Reply:
Tell me if I'm wrong but the text is of type `const char*` and thus, it shouldn't be freed since it is constant (?).

Reply:
If the data comes from `cobc_strdup`, then it is allocated (permanently, not cached), not a constant.
`const char *` may be a wrong type.